### PR TITLE
[FEATURE] White BG during 2D export in TOPPView.

### PR DESCRIPTION
**Pull request by jpfeuffer**
# Description

White BG during 2D export in TOPPView.
test

Also, fixes assert errors introduced in recent TOPPView

# Checklist:
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] New and existing unit tests pass locally with my changes
- [X] Updated or added python bindings for changed or new classes. (Tick if no updates were necessary.)

### CODE REVIEW

**Comment by timosachsenberg** [comment]
What about the 3rd party changes?

File: src/openms_gui/source/VISUAL/SpectrumWidget.cpp [hunk 1]
```diff
@@ -224,6 +224,7 @@ namespace OpenMS
   void SpectrumWidget::saveAsImage()
   {
     QString file_name = QFileDialog::getSaveFileName(this, "Save File", "", "Images (*.bmp *.png *.jpg *.gif)");
+    this->setStyleSheet("background: white");
```
Comment (timosachsenberg) [inline src/openms_gui/source/VISUAL/SpectrumWidget.cpp:227]: maybe store old stylesheet? and reset below?

File: src/openms_gui/source/VISUAL/SpectrumWidget.cpp [hunk 2]
```diff
@@ -354,4 +356,12 @@ namespace OpenMS
     event->acceptProposedAction();
   }
 
+  void SpectrumWidget::paintEvent(QPaintEvent * /*event*/)
+  {
+    QStyleOption opt;
+    opt.init(this);
+    QPainter p(this);
+    style()->drawPrimitive(QStyle::PE_Widget, &opt, &p, this);
```
Comment (timosachsenberg) [inline src/openms_gui/source/VISUAL/SpectrumWidget.cpp:364]: add comment on what this does?

Comment (jpfeuffer) [reply src/openms_gui/source/VISUAL/SpectrumWidget.cpp:364]: I don't know what it does. The docu said I need it.

**Comment by timosachsenberg** [comment]
Looks good from my side.

File: src/openms_gui/source/VISUAL/SpectrumWidget.cpp (hunk 1 above)
Comment (cbielow) [reply src/openms_gui/source/VISUAL/SpectrumWidget.cpp:229]: Suggestion: RAIICleanup for restoring old stylesheet

File: src/openms_gui/source/VISUAL/APPLICATIONS/TOPPViewBase.cpp [hunk 3]
```diff
@@ -1891,7 +1891,7 @@ namespace OpenMS
   {
     LayerData& layer = getActiveCanvas()->getCurrentLayer();
     LayerAnnotatorAMS annotator;
-    assert(log != nullptr);
+    assert(log_ != nullptr);
```
Comment (cbielow) [inline src/openms_gui/source/VISUAL/APPLICATIONS/TOPPViewBase.cpp:1894]: Ouch. Thanks

**Review by cbielow — COMMENTED** [review]
Can you document why this is needed?

**Pull request merged**